yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -819,12 +819,22 @@ private[spark] class Client(
       javaOpts += "-XX:CMSIncrementalDutyCycle=10"
     }
 
-    // Include driver-specific java options if we are launching a driver
-    if (isClusterMode) {
```

Comment:
For client mode I would rather move this check somewhere more generic in order to keep it consistent across cluster managers.   Right now (without this patch) it doesn't error if you specify Xmx in the java options running in yarn client mode, the same thing is true for standalone mode.

So I think we should either just change this back and only check in cluster mode to keep previous behavior (and its consistent with standalone mode) or we should move the check into SparkClassCommandBuilder and SparkSubmitCommandBuilder so they are applied for all cluster managers and it would check before it ever launches them.

I'm leaning towards the latter and move them into the CommandBuilder classes. @dhruve  can you look into adding a check there?